test/module/libs/cache/transaction_cache_test.cpp

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include <vector>
+
+#include "cache/collection_set.hpp"
+
+using namespace iroha::set;
+
+/**
+ * @given empty set
+ * @when  check that empty set doesn't contain elements
+ * AND insert some collection
+ * @then  check that elements are appeared
+ */
+TEST(TransactionCacheTest, insert) {
+  auto number_of_calls = 0;
+  CollectionSet<int> set;
+  set.foreach ([&number_of_calls](const auto &val) { number_of_calls++; });
+  ASSERT_EQ(0, number_of_calls);
+
+  set.insertValues(std::vector<int>({1, 2}));
+
+  set.foreach ([&number_of_calls](const auto &val) { number_of_calls++; });
+  ASSERT_EQ(2, number_of_calls);
+}
+
+/**
+ * @given empty set
+ * @when insert some collection
+ * AND insert duplicated
+ * @then  check that duplicates are not appeared
+ */
+TEST(TransactionCacheTest, insertDuplicates) {
+  auto number_of_calls = 0;
```

Comment:
here is the number of elements, not calls

Reply:
Number of calls, because we count lambda invocations on `foreach`. But the numbers are the same.